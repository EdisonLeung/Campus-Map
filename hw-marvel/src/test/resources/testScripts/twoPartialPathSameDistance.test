# Test pathfinding on a node with 2 paths that have the same distance but the two
# paths also share a same path for part of the traversal. Tests if the least
# lexicographical path is chosen

# Create graph
CreateGraph partialPath

# Add nodes
AddNode partialPath a
AddNode partialPath b
AddNode partialPath c

# Add edges
AddEdge partialPath a b ABa
AddEdge partialPath a b ABb
AddEdge partialPath b c BC

# test pathfinding from a to c
FindPath partialPath a c