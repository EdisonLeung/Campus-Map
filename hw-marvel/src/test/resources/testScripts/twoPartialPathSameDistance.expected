# Test pathfinding on a node with 2 paths that have the same distance but the two
# paths also share a same path for part of the traversal. Tests if the least
# lexicographical path is chosen

# Create graph
created graph partialPath

# Add nodes
added node a to partialPath
added node b to partialPath
added node c to partialPath

# Add edges
added edge ABa from a to b in partialPath
added edge ABb from a to b in partialPath
added edge BC from b to c in partialPath

# test pathfinding from a to c
path from a to c:
a to b via ABa
b to c via BC